tests/reference_tests/reference_projects_test.py

```diff
@@ -0,0 +1,258 @@
+"""Test suite for running FABulous on reference projects with difference checking.
+
+This module tests FABulous against reference projects defined in a config file,
+supporting both "run" mode (error checking) and "diff" mode (regression testing).
+"""
+
+import os
+import shutil
+from pathlib import Path
+from typing import Any, Literal, NamedTuple
+
+import pytest
+import yaml
+from loguru import logger
+
+from FABulous.FABulous_CLI.FABulous_CLI import FABulous_CLI
+from FABulous.FABulous_CLI.helper import (
+    setup_logger,
+)
+from tests.CLI_test.conftest import normalize, run_cmd
+from tests.reference_tests.helpers import (
+    compare_directories,
+    format_file_differences_report,
+)
+
+
+class ReferenceProject(NamedTuple):
+    """Configuration for a reference project test."""
+
+    name: str
+    path: Path
+    language: Literal["verilog", "vhdl"]
+    test_mode: Literal["run", "diff"]
+    description: str = ""
+    expected_outputs: list[str] | None = None
+    include_patterns: list[str] | None = None
+    exclude_patterns: list[str] | None = None
+    commands: list[str] | None = None
+    skip_reason: str | None = None
+
+
+def load_reference_projects_config(config_path: Path) -> list[ReferenceProject]:
+    """Load reference projects configuration from YAML file."""
+    if not config_path.exists():
+        raise FileNotFoundError(f"Config file not found: {config_path}")
+
+    with config_path.open("r") as f:
+        config_data = yaml.safe_load(f)
+
+    projects = []
+    for project_data in config_data.get("reference_projects", []):
+        if Path(project_data["path"]).is_absolute():
+            path = Path(project_data["path"])
+        else:
+            path = config_path.parent / project_data["path"]
+        try:
+            project = ReferenceProject(
+                name=project_data["name"],
+                path=path.resolve(),
+                language=project_data["language"],
+                test_mode=project_data["test_mode"],
+                description=project_data.get("description", ""),
+                expected_outputs=project_data.get("expected_outputs"),
+                include_patterns=project_data.get("include_patterns"),
+                exclude_patterns=project_data.get("exclude_patterns"),
+                commands=project_data.get("commands"),
+                skip_reason=project_data.get("skip_reason"),
+            )
+            projects.append(project)
+        except KeyError as e:
+            logger.warning(f"Invalid project config, missing key {e}: {project_data}")
+        except Exception as e:  # noqa: BLE001
+            logger.warning(f"Failed to load project config: {e}")
+
+    return projects
+
+
+def run_fabulous_commands_with_logging(
```

Comment:
I think this in helper make more sense.

Maybe we can merge this with the `cli` fixture such that the `cli` fixture is a generator fixture which return a `cli` factory, and we can pass argument into the to the factory to get a configured CLI.

Reply:
After a bit more thinking, this also means the `CLI_test` will need to be updated as well. Could be done in  the future